### ISSUE TITLE
Split command logic into a separate package

### ISSUE THREAD

**Issue by Blokje5** [open-policy-agent/conftest#316]
Right now most of the logic resides with the commands. E.g. with the Test command, most of the logic to fetch the related configuration files and query Rego policies is stored in functions in [the commands package](https://github.com/open-policy-agent/conftest/blob/master/internal/commands/test.go). However this approach has two downsides:

1. Command Configuration & CLI interaction (e.g. viper calls) are intermixed with the actual logic of the command (querying Rego policies, fetching configuration files), making it hard to test and reuse the logic.
2. Due to the commands being stored in internal/commands this logic can't be imported into other applications. This could be relevant if people want to write their own plugins.

Instead I would like to propose the following: we split the CLI interaction (which can remain in internal/commands) from the actual logic in the commands. The actual logic can reside in a seperate package e.g. `runner` or `action`. For each command we would have a "runner", essentially a struct that holds the required configuration options and has a `Run` method, e.g.:

```
type TestRunner struct {
  ConfigurationFiles []string
  PolicyDir []string

  // Other configuration options
}

func (t *TestRunner) Run(ctx context.Context) error {
  // Actual logic in here
}
```

This `Run` method would be called from within the commands. This approach has the following benefits:

1. It is easier to create unit tests for the commands in go as you would not have to deal with flag binding and exit codes.
2. The runners can be reused by people wanting to import that logic into their own go applications as we can expose it externally. This does mean we have to maintain the interface so we can start with moving the runners into an internal package until we feel comfortable having the interface exposed.

**Comment by boranx** [comment]
I agree with @Blokje5 
The viper specific blocks and actual logic should be separated and also `conftest` should be able to scalable in terms of new commands which can be added in the future. I liked the `Run()` idea which can be called from several commands. The `parse` and `test` might be good candidates that can adapt `Run`.
This will help us to write more unit tests to cover rather than black-box testing as well. 
Considering this as a refactor epic, we can start creating issues, but first, we all have to be ok
@jpreese any thoughts?

**Comment by Blokje5** [comment]
@boranx I can also create a WIP PR soon, might be nice to have a discussion there as well on the new structure. I already changed a few commands locally (without changing too much, I want to avoid breaking things). I can create a WIP PR for the few commands I changed locally.

**Comment by boranx** [comment]
That would be great! Having a POC on it will turn out the discussions be concrete...

**Comment by Blokje5** [comment]
@boranx I pushed a branch where you can see the general idea.

**Comment by jpreese** [comment]
Resolved via #327. Thanks @Blokje5! :tada:
